src/main/java/com/google/cloud/tools/managedcloudsdk/install/Downloader.java

```diff
@@ -48,14 +50,18 @@
 
   /** Download an archive, this will NOT overwrite a previously existing file. */
   public void download() throws IOException, InterruptedException {
+    download(new TextBarFactory());
+  }
+
+  @VisibleForTesting
+  void download(TextBarFactory textBarFactory) throws IOException, InterruptedException {
```

Comment:
This API is very strange and feels like it's mixing things that shouldn't be mixed. I would not expect that a download method requires a piece of UI like textBarFactory.

Reply:
it's not part of the API, it's just a way to inject this.